README.md

```diff
@@ -1,20 +1,214 @@
+  end
+```
+
+Okay, well, surely it can't be that simple. Let's load up our Rails
+server and browse to `/posts`. Click one of the "More" buttons, and,
+just like that, it updates the post body. We didn't have to change a
+thing. Think of it as a testament to how great a job we did writing our
+own serializer.
+
+#### Including Associations
+
+Now if we click on the first post and use our `Next...` link, that
+should mostly work too.
+
+I say mostly, because it's not updating the author name. That's
+something we added in to our serializer, but by default, `to_json` only
+serializes the main object, not any associations. How can we change
+that?
+
+You can tell `to_json` what associated objects to include, using the
+`include` option.
+
+```ruby
+# posts_controller.rb
+# ...
+  def post_data
+    post = Post.find(params[:id])
+    render json: post.to_json(include: :author)
+  end
+```
+
+Now if we reload that post show page and click `Next`, the author should
+update as well.
+
+#### Only Render The Data We Need
+
+If we browse to `/posts/id/post_data`, we can see the raw JSON of our
+object. It should look something like this:
+
+```javascript
+{
+  id: 1,
+  title: "A Blog Post By Stephen King",
+  description: "This is a blog post by Stephen King. It will probably be a movie soon.",
+  created_at: "2016-02-22T00:29:21.022Z",
+  updated_at: "2016-02-22T00:29:21.022Z",
+  author_id: 1,
+    author: {
+      id: 1,
+      name: "Stephen King",
+      hometown: null,
+      created_at: "2016-02-22T00:29:20.999Z",
+      updated_at: "2016-02-22T00:29:20.999Z"
+    }
+}
+```
+
+**Note:** This would be a great time to install [JSONView](https://chrome.google.com/webstore/detail/jsonview/chklaanhfefbnpoihckbnefhakgolnmc?hl=en) if you haven't already!
+
+It's great that `to_json` gives us all this, but we don't really need
+all of it.
+
+A good API endpoint should return *only* the data that is needed,
+nothing more. So how do we get rid of that stuff?
+
+It turns out `to_json` gives us ways to exclude data as well, using the
+`only` option, similar to how we'd specify certain routes for a resoure.
+
+```ruby
+# posts_controller.rb
+# ...
+  def post_data
+    post = Post.find(params[:id])
+    #render json: PostSerializer.serialize(post)
+    render json: post.to_json(only: [:title, :description, :id],
+                              include: [ author: { only: [:name]}])
+  end
+```
+
+We can use `only` both on the main object and the included objects.
+
+**Top-tip:** Notice that we have to pass `author:` inside an array
+for `include` now that we are specifying additional options.
+
+Reloading the `/posts/id/post_data` page now gives us something more
+like this:
+
+```javascript
+{
+  id: 1,
+  title: "A Blog Post By Stephen King",
+  description: "This is a blog post by Stephen King. It will probably be a movie soon.",
+  author: {
+    name: "Stephen King"
+  }
+}
+```
+
+Which is exactly the data we need to get the job done.
```

Comment:
This section does a really nice job of walking the user through the code and how they might want to use `to_json`.